### PR TITLE
Prior to PHP 5.5, empty() only supports variables

### PR DESCRIPTION
**Pull request by bjork**
Test if the parameter of empty() is an arbitrary expression, and not
just a variable.

### CODE REVIEW

File: src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php [hunk 1]
```diff
@@ -332,6 +332,9 @@ public function leaveNode(Node $node)
 
         } elseif ($node instanceof Node\Scalar\MagicConst) {
             $this->computeConstantVersions($node, $node->getName());
+
+        } else if ( $node instanceof Node\Expr\Empty_ ) {
```
Comment (GrahamCampbell) [inline src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php]: elseif not else if

File: src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php [hunk 2]
```diff
@@ -1129,6 +1132,22 @@ private function computePhpFeatureVersions(Node $node)
         } elseif ($node instanceof Node\Stmt\Goto_) {
             $versions = array('php.min' => '5.3.0');
             $this->updateLocalVersions($versions);
+
+        } else if ( $node instanceof Node\Expr\Empty_ ) {
```
Comment (GrahamCampbell) [inline src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php]: cs

File: src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php [hunk 3]
```diff
@@ -1129,6 +1132,22 @@ private function computePhpFeatureVersions(Node $node)
...
+
+            // If the parameter of empty() is an arbitrary expression,
+            // and not just a variable.
+            if ( $node->expr instanceof Node\Expr
```
Comment (GrahamCampbell) [inline src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php]: cs

**Comment by llaville** [comment]
For documentation, here is the link http://www.php.net/manual/en/migration55.new-features.php#migration55.new-features.empty

I need to create a test unit for quality assurance. Will do it tomorrow

**Comment by llaville** [comment]
Release 4.4.1 is planned for saturday morning (July 18). 
Any feedback are welcome before deadline !

**Pull request merged**